contracts/relayer/RelayerAragonAppWithParameterizedSender.sol

```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.4.24;
+
+import "./BaseRelayer.sol";
+
+
+contract RelayerAragonAppWithParameterizedSender is BaseRelayer {
+    modifier relayedAuth(address _sender, bytes32 _role) {
+        require(canPerform(_sender, _role, new uint256[](0)), ERROR_AUTH_FAILED);
+        _;
+    }
+
+    modifier relayedAuthP(address _sender, bytes32 _role, uint256[] _params) {
+        require(canPerform(_sender, _role, _params), ERROR_AUTH_FAILED);
+        _;
+    }
+
+    function exec(address from, uint256 nonce, bytes calldata, bytes signature) external refundGas auth(OFF_CHAIN_RELAYER_SERVICE_ROLE) {
+        assertValidTransaction(from, nonce, calldata, signature);
+
+        setUsedNonce(from, nonce, true);
+        bool success = address(this).call(calldata);
```

Comment:
Are this and the next one supposed to be the cases in the first column of the description? The "on-chain relayers"? In that case, shouldn't they call to an external app, as they are just a "middleman", instead to itself?

Reply:
No, this example would be the app relayer, i.e. an AragonApp including relayer logic. It is forwarding the call  to itself to enter the target function actually.